File: internal/translator/schema/schema.go
// Package schema provides shared protocol constants and message types for protocol translation.
package schema

// Standard role constants.
const (
	RoleSystem    = "system"
	RoleUser      = "user"
	RoleAssistant = "assistant"
	RoleTool      = "tool"
	RoleDeveloper = "developer"
)

// Gemini role constants.
const (
	GeminiRoleUser  = "user"
	GeminiRoleModel = "model"
)

// OpenAI content block type constants.
const (
	OpenaiBlockText       = "text"
	OpenaiBlockImageURL   = "image_url"
	OpenaiBlockImage      = "image"
	OpenaiBlockInputAudio = "input_audio"
	OpenaiBlockAudioURL   = "audio_url"
	OpenaiBlockFile       = "file"
	OpenaiBlockFunction   = "function"
)

// Claude content block type constants.
const (
	ClaudeBlockText             = "text"
	ClaudeBlockImage            = "image"
	ClaudeBlockDocument         = "document"
	ClaudeBlockToolUse          = "tool_use"
	ClaudeBlockToolResult       = "tool_result"
	ClaudeBlockThinking         = "thinking"
	ClaudeBlockRedactedThinking = "redacted_thinking"
)

// Responses API item type constants.
const (
	ResponsesItemMessage            = "message"
	ResponsesItemFunctionCall       = "function_call"
	ResponsesItemFunctionCallOutput = "function_call_output"
	ResponsesItemReasoning          = "reasoning"
	ResponsesItemOutputText         = "output_text"
	ResponsesItemInputText          = "input_text"
	ResponsesItemInputImage         = "input_image"
	ResponsesItemSummaryText        = "summary_text"
)

// OpenAI finish reason constants.
const (
	OpenaiFinishStop          = "stop"
	OpenaiFinishLength        = "length"
	OpenaiFinishToolCalls     = "tool_calls"
	OpenaiFinishContentFilter = "content_filter"
)

// Claude stop reason constants.
const (
	ClaudeStopEndTurn      = "end_turn"
	ClaudeStopMaxTokens    = "max_tokens"
	ClaudeStopToolUse      = "tool_use"
	ClaudeStopStopSequence = "stop_sequence"
)

// Gemini finish reason constants.
const (
	GeminiFinishStop              = "STOP"
	GeminiFinishMaxTokens         = "MAX_TOKENS"
	GeminiFinishSafety            = "SAFETY"
	GeminiFinishRecitation        = "RECITATION"
	GeminiFinishBlocklist         = "BLOCKLIST"
	GeminiFinishProhibitedContent = "PROHIBITED_CONTENT"
)

// ValidOpenaiContentTypes specifies valid content types in OpenAI message parts.
var ValidOpenaiContentTypes = map[string]bool{
	"text":        true,
	"image_url":   true,
	"image":       true,
	"input_audio": true,
	"audio_url":   true,
	"file":        true,
}

// ValidOpenaiMessageTypes specifies valid message types for OpenAI messages.
var ValidOpenaiMessageTypes = map[string]bool{
	"text":        true,
	"image_url":   true,
	"image":       true,
	"tool_calls":  true,
	"tool_result": true,
}

// Model fallback and default values.
const (
	ModelFallback    = "unknown"
	DefaultImageMime = "image/png"
)

File: open-sse/src/translator/schema.rs
//! Schema constants — mirror of open-sse/translator/schema/

pub mod role {
    pub const USER: &str = "user";
    pub const ASSISTANT: &str = "assistant";
    pub const TOOL: &str = "tool";
    pub const SYSTEM: &str = "system";
    pub const DEVELOPER: &str = "developer";
}

pub mod gemini_role {
    pub const USER: &str = "user";
    pub const MODEL: &str = "model";
}

pub mod openai_block {
    pub const TEXT: &str = "text";
    pub const IMAGE_URL: &str = "image_url";
    pub const IMAGE: &str = "image";
    pub const INPUT_AUDIO: &str = "input_audio";
    pub const AUDIO_URL: &str = "audio_url";
    pub const FILE: &str = "file";
    pub const FUNCTION: &str = "function";
}

pub mod claude_block {
    pub const TEXT: &str = "text";
    pub const IMAGE: &str = "image";
    pub const DOCUMENT: &str = "document";
    pub const TOOL_USE: &str = "tool_use";
    pub const TOOL_RESULT: &str = "tool_result";
    pub const THINKING: &str = "thinking";
    pub const REDACTED_THINKING: &str = "redacted_thinking";
}

pub const DEFAULT_MAX_TOKENS: u64 = 64000;
pub const DEFAULT_MIN_TOKENS: u64 = 32000;
